### PR TITLE
Parsing CharacterSet to exclude single and double quotes for postgres

### LINKED ISSUE #23885: IllegalCharsetNameException: 'utf-8' occurs while connecting to ShardingSphere using sqlalchemy asyncpg
## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?
ShardingSphere Proxy - 5.1.2

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy

### Expected behavior
By using sqlalchemy asyncpg we should be able to connect to ShardingSphere with the connection string once ShardingSphere proxy is started. 
### Actual behavior
Not able to connect to ShardingSphere because IllegalCharsetNameException is coming.
### Reason analyze (If you can)
```
[ERROR] 2023-02-01 03:26:02.303 [epollEventLoopGroup-3-2] o.a.s.p.f.n.FrontendChannelInboundHandler - Exception occur: 
java.nio.charset.IllegalCharsetNameException: 'utf-8'
	at java.base/java.nio.charset.Charset.checkName(Charset.java:308)
	at java.base/java.nio.charset.Charset.lookup2(Charset.java:482)
	at java.base/java.nio.charset.Charset.lookup(Charset.java:462)
	at java.base/java.nio.charset.Charset.forName(Charset.java:526)
	at org.apache.shardingsphere.proxy.backend.text.admin.postgresql.PostgreSQLCharacterSets.findCharacterSet(PostgreSQLCharacterSets.java:111)
	at org.apache.shardingsphere.proxy.frontend.postgresql.authentication.PostgreSQLAuthenticationEngine.processStartupMessage(PostgreSQLAuthenticationEngine.java:90)
	at org.apache.shardingsphere.proxy.frontend.postgresql.authentication.PostgreSQLAuthenticationEngine.authenticate(PostgreSQLAuthenticationEngine.java:83)
	at org.apache.shardingsphere.proxy.frontend.netty.FrontendChannelInboundHandler.authenticate(FrontendChannelInboundHandler.java:83)
	at org.apache.shardingsphere.proxy.frontend.netty.FrontendChannelInboundHandler.channelRead(FrontendChannelInboundHandler.java:75)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:327)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:299)
	at io.netty.handler.codec.ByteToMessageCodec.channelRead(ByteToMessageCodec.java:103)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:795)
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:480)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:829)

```
### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
`sh start.sh`

### Example codes for reproduce this issue (such as a github link).
```import asyncio
import sys
from sqlalchemy.ext.asyncio import create_async_engine
from sqlalchemy.future import select
from sqlalchemy.sql import text

async def async_main():
    engine = create_async_engine(
        "postgresql+asyncpg://root:root@127.0.0.1:3307/hint_db",
        echo=True,
    )
    print(sys.getdefaultencoding())
    async with engine.connect() as conn:
        result = await conn.execute(select(text('tracking_id')))
        print(result)

asyncio.run(async_main())

### PR DESCRIPTION
**Pull request by jyotisharma7**
Fixes #23885 
Changes proposed in this pull request:
  - The issue caused by the single quote in utf-8 while connecting to ShardingSphere for postgres is fixed.

---

Before committing this PR, I'm sure that I have checked the following options:
- [X ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [X ] I have self-reviewed the commit code.
- [X ] I have (or in comment I request) added corresponding labels for the pull request.
- [ X] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [X ] I have made corresponding changes to the documentation.
- [ X] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/handler/admin/postgresql/PostgreSQLCharacterSets.java [hunk 1]
```diff
@@ -107,7 +107,18 @@ public enum PostgreSQLCharacterSets {
...
+        PostgreSQLCharacterSets result = CHARACTER_SETS_MAP.get(formattedCharsetName.toUpperCase());
+        return null != result && null != result.charset ? result.charset : Charset.forName(formattedCharsetName);
+    }
+    
+    /**
+     * Find corresponding {@link Charset} by charset name defined in PostgreSQL.
+     *
+     * @param value that needs to be formatted
+     * @return formatted string without any single/double quotes {@link String}
+     */
```
Comment (TeslaCN) [inline proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/handler/admin/postgresql/PostgreSQLCharacterSets.java:120]: Javadoc is unnecessary here.

File: proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/handler/admin/postgresql/executor/PostgreSQLSetCharsetExecutor.java [hunk 2]
```diff
@@ -34,7 +34,7 @@ public final class PostgreSQLSetCharsetExecutor implements PostgreSQLSessionVari
     
     @Override
     public void handle(final ConnectionSession connectionSession, final String variableName, final String assignValue) {
-        String value = formatValue(assignValue.trim());
+        String value = assignValue.trim();
         connectionSession.getAttributeMap().attr(CommonConstants.CHARSET_ATTRIBUTE_KEY).set(parseCharset(value));
```
Comment (TeslaCN) [inline proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/handler/admin/postgresql/executor/PostgreSQLSetCharsetExecutor.java:38]: The variable `value` could be inlined.

File: proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/handler/admin/postgresql/executor/PostgreSQLSetCharsetExecutor.java [hunk 3]
```diff
@@ -34,7 +34,7 @@ public final class PostgreSQLSetCharsetExecutor implements PostgreSQLSessionVari
     
     @Override
     public void handle(final ConnectionSession connectionSession, final String variableName, final String assignValue) {
-        String value = formatValue(assignValue.trim());
```
Comment (TeslaCN) [inline proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/handler/admin/postgresql/executor/PostgreSQLSetCharsetExecutor.java:37]: Consider removing the method `formatValue` if it is unused.

Comment (surukonda) [reply proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/handler/admin/postgresql/executor/PostgreSQLSetCharsetExecutor.java:37]: Ah! that was miss, it is removed now in the updated PR.

File: proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/handler/admin/postgresql/executor/PostgreSQLSetCharsetExecutor.java (hunk 2 above)
Comment (surukonda) [reply proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/handler/admin/postgresql/executor/PostgreSQLSetCharsetExecutor.java]: Done!

File: proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/handler/admin/postgresql/PostgreSQLCharacterSets.java (hunk 1 above)
Comment (surukonda) [reply proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/handler/admin/postgresql/PostgreSQLCharacterSets.java]: Removed

**Review by TeslaCN — APPROVED** [review]
Thank you @jyotisharma7

**Comment by surukonda** [comment]
Thanks @TeslaCN for approving and merging it, any idea in which version this fix will be available? @jyotisharma7 and I work part of same team.

**Comment by TeslaCN** [comment]
> Thanks @TeslaCN for approving and merging it, any idea in which version this fix will be available? @jyotisharma7 and I work part of same team.

The fix will be available in next release. But I'm not sure when for now. Please follow our community.

**Pull request merged**